tests/Fixer/FunctionNotation/StaticLambdaFixerTest.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\FunctionNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\FunctionNotation\StaticLambdaFixer
+ */
+final class StaticLambdaFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string $expected
+     * @param string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            'sample' => [
+                "<?php\n\$a = static function () use (\$b)\n{   echo \$b;\n};",
+                "<?php\n\$a = function () use (\$b)\n{   echo \$b;\n};",
+            ],
+            'minimal double fix case' => [
+                '<?php $a=static function(){};$b=static function(){};',
+                '<?php $a=function(){};$b=function(){};',
+            ],
+            [
+                '<?php $a  /**/  =   /**/     static function(){};',
+                '<?php $a  /**/  =   /**/     function(){};',
+            ],
+            [
+                '<?php $a  /**/  =   /**/ static function(){};',
+                '<?php $a  /**/  =   /**/ function(){};',
+            ],
+            [
+                '<?php $a  /**/  =   /**/static function(){};',
+                '<?php $a  /**/  =   /**/function(){};',
+            ],
+        ];
+    }
+
+    /**
+     * @param string $expected
+     *
+     * @dataProvider provideDoNotFixCases
+     */
+    public function testDoNotFix($expected)
+    {
+        $this->doTest($expected);
+    }
+
+    public function provideDoNotFixCases()
+    {
+        return [
+            [
+                '<?php
+                    class A
+                    {
+                        public function B()
+                        {
+                            $a = function () {
+                                $b = \'this\';
```

Comment:
What about the use of `$this` inside an `eval`?

Reply:
good ol' eval... 
fair point though :+1: , I can update to check if `eval` is used within the lambda

Reply:
`eval` can't be handled: how would you detect `eval(' return ' . strrev('; siht$ '));`?

Mark the fixer risky and give up the eval.

Reply:
if skipping `eval` in general would make fixer non-risky, then don't handle `eval` at all.
or even more - if we need some tricks to hande eval, let us not do it. we don't need to support crap

Reply:
The fixer will skip the lambda if `eval` is used within the body of the lambda,
same as for example when `require_once` is used etc.
This makes the fixer far more save to use, but at the small cost that some cases are not fixed while it could be. Than again, if you use `eval` (in a lambda) than you won't find much help from other tooling like SCA, basically you're on your own with stuff like that ;)

Reply:
@SpacePossum I would find useful to add your last thoughts in the extended description.